src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,52 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+
     public void getStatistic(String fromFileName, String toFileName) {
+        StringBuilder report = createReport(fromFileName);
+        writeToFile(report, toFileName);
+    }
+
+    private void writeToFile(StringBuilder report, String toFileName) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName))) {
+            writer.write(report.toString());
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write to file" + e);
+        }
+    }
 
+    private StringBuilder createReport(String fromFileName) {
+        StringBuilder report = new StringBuilder();
+        int supplyTotal = 0;
+        int buyTotal = 0;
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            String value;
+            while (true) {
+                line = reader.readLine();
+                if (line == null) {
+                    break;
+                } else if (line.contains("supply")) {
+                    value = line.split(",")[1];
+                    supplyTotal += Integer.parseInt(value);
+                } else {
+                    value = line.split(",")[1];
+                    buyTotal += Integer.parseInt(value);
+                }
+            }
```

Comment:
Perpetual loop is bad practice.
It is better to set a condition for better readability
```suggestion
            line = reader.readLine();
            while (line != null) {
                if (line.contains("supply")) {
                    value = line.split(",")[1];
                    supplyTotal += Integer.parseInt(value);
                } else {
                    value = line.split(",")[1];
                    buyTotal += Integer.parseInt(value);
                }
                line = reader.readLine();
            }
```